sdk/provisioning/Azure.Provisioning/src/storage/StorageAccount.cs

```diff
@@ -53,27 +50,7 @@ public StorageAccount(IConstruct scope, StorageKind kind, StorageSkuName sku, Re
         /// <inheritdoc/>
         protected override string GetAzureName(IConstruct scope, string resourceName)
         {
-            var span = resourceName.AsSpan();
-            StringBuilder stringBuilder = new StringBuilder();
-            for (int i = 0; i < span.Length; i++)
-            {
-                char c = span[i];
-                if (!char.IsLetterOrDigit(c))
-                {
-                    continue;
-                }
-                if (char.IsLetter(c))
-                {
-                    stringBuilder.Append(char.ToLowerInvariant(c));
-                }
-                else
-                {
-                    stringBuilder.Append(c);
-                }
-            }
-            stringBuilder.Append(Guid.NewGuid().ToString("N"));
-
-            return stringBuilder.ToString(0, Math.Min(stringBuilder.Length, 24));
+            return $"toLower(take(concat('{resourceName}', uniqueString(resourceGroup().id)), 24))";
```

Comment:
probably should just remove the guid from this and make resources that need uniqueness implement that in their overload.

Feels bad to have all names turn into functions

Reply:
this is in the StorageAccount only